l2gv2/models.py

```diff
@@ -150,83 +255,146 @@ def Node2Vec_(data, emb_dim, w_length=20, c_size=10,
+            )
+        )
+
     return patch_list
 
 
-def chunk_embedding(chunk_size, patches, dim=2):  #this only work for Autonomous System dataset. 
+def chunk_embedding(chunk_size, patches, dim=2):
+    """Compute the embeddings of a list of patches using VGAE and align them. 
+    This only works for Autonomous System dataset.
+
+    Args:
+
+        chunk_size: size of the chunks.
+
+        patches: list of torch_geometric.data.Data objects.
+
+        dim: dimension of the embeddings, default is 2.
+
+    Returns:
+
+        Nodes_emb: embeddings of the nodes.
 
-    tAll=[from_networkx(g) for g in patches]
-    list_set_nodes=[set(patches[i].nodes()) for i in range(len(patches))]
+        Prob: WeightedAlignmentProblem object.
 
-    nodes_in_intersection=set.intersection(*[set(gr) for gr in list_set_nodes])
-    
-    nodes_in_union=set()
+    """
+
+    t_all = [from_networkx(g) for g in patches]
+    list_set_nodes = [set(patches[i].nodes()) for i in range(len(patches))]
+
+    nodes_in_intersection = set.intersection(*[set(gr) for gr in list_set_nodes])
+
+    nodes_in_union = set()
     for s in list_set_nodes:
         nodes_in_union = nodes_in_union.union(s)
-    
-    missing_nodes=set(i for i in range(max(nodes_in_union)))
+
+    missing_nodes = set(i for i in range(max(nodes_in_union)))
     for i in range(max(nodes_in_union)):
         if i in nodes_in_union:
             missing_nodes.remove(i)
-    
-    for i, p in tqdm(enumerate(tAll)):
-        p.nodes=torch.Tensor(list(patches[i].nodes)).int()
-        p.num_nodes=p.nodes.size(0)
-    sub_patches = [tAll[i:i + chunk_size] for i in range(0, len(tAll), chunk_size)]
-
-    nodes_in_intersection=[]
-    list_nodes_tot=[]
+
+    for i, p in tqdm(enumerate(t_all)):
+        p.nodes = torch.Tensor(list(patches[i].nodes)).int()
+        p.num_nodes = p.nodes.size(0)
+    sub_patches = [t_all[i : i + chunk_size] for i in range(0, len(t_all), chunk_size)]
+
+    nodes_in_intersection = []
+    list_nodes_tot = []
     for g in tqdm(sub_patches):
-    
-        nodes_in_each_p=[set(p.nodes.tolist()) for p in g]
+        nodes_in_each_p = [set(p.nodes.tolist()) for p in g]
         nodes_in_intersection.append(set.intersection(*nodes_in_each_p))
-        nodes_tot=[]
+        nodes_tot = []
         for p in nodes_in_each_p:
-            nodes_tot+=p
-        list_nodes_tot.append(set(nodes_tot))  
-
-    emb=[]
-    Ppatch=[]
-    for i, As in tqdm(enumerate(sub_patches)):
-        p_emb=VGAE_patch_embeddings(As, dim=dim, hidden_dim=32, num_epochs=100, decoder=None, device='cpu', lr=0.01)
-    
-        prob = l2g.utils.WeightedAlignmentProblem(p_emb[0])  #embedding of the full graph using embeddings of each patch
-    
+            nodes_tot += p
+        list_nodes_tot.append(set(nodes_tot))
+
+    emb = []
+    for i, patch in tqdm(enumerate(sub_patches)):
+        p_emb = vgae_patch_embeddings(
+            patch,
+            dim=dim,
+            hidden_dim=32,
```

Comment:
no device=cpu here

Reply:
@abhidg This function had a parameter device which I deleted because there is also the 

> device = torch.device("cuda" if torch.cuda.is_available() else "cpu")

in the code which sets the device. Do you think I should add back the parameter to allow overriding the device at when the function is called?